src/Raven.Client/Exceptions/Documents/CollectionMismatchException.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace Raven.Client.Exceptions.Documents
+{
+    public class CollectionMismatchException : RavenException
+    {
+        public string Id { get; set; }
+        public CollectionMismatchException()
```

Comment:
create set of standard ctors for exceptions and ThrowFor method (check IndexDoesNotExistException)